### ISSUE TITLE
almost. just a 

### ISSUE THREAD

**Issue by transfluxus** [samim23/Novelty-Search-Live#1]
Hi, 
tried it, but after getting through the installation I get when running it, or someone else might get

```
Ableton Live Novelty Search - Song Name: save/mysong
*** timed out waiting for server response
Traceback (most recent call last):
  File "noveltysearchlive.py", line 69, in <module>
    set.scan(scan_devices = True,scan_clip_names = True,)
  File "/Users/raminsoleymani/Downloads/Novelty-Search-Live-master/venv/lib/python2.7/site-packages/live/set.py", line 590, in scan
    track_count = self.num_tracks
  File "/Users/raminsoleymani/Downloads/Novelty-Search-Live-master/venv/lib/python2.7/site-packages/live/set.py", line 319, in num_tracks
    return self.live.query("/live/tracks")[0]
IndexError: list index out of range
```

pylive requires  the ableton live plugin (or whatever it's called) LiveOSC. pylive links to a project page, which is quite outdated. one needs LiveOSC2, which works for live 9:
https://github.com/stufisher/LiveOSC2

still I'm getting
```
  File "/Users/cooluser/Novelty-Search-Live-master/venv/lib/python2.7/site-packages/live/query.py", line 68, in __init__
    self.osc_server = liblo.Server(listen_port)
  File "src/liblo.pyx", line 602, in liblo.Server.__init__ (src/liblo.c:8305)
liblo.ServerError: server error 9904: cannot find free port
```
... but I guess I'm close. Do you have an idea what could go wrong with ableton or that python library?

for other people. under mac there might be more things that need to be installed. for example liblo with
`brew install liblo`

It should probably also noted that this works only in python 2.?

**Comment by samim23** [comment]
I followed the installation instructions at https://github.com/ideoforms/pylive. 
Installed LiveOSC (http://livecontrol.q3f.org/ableton-liveapi/liveosc/) & enabled it as a Control Surface in Ableton Live. But since I did this done this quite some time ago, might have changed.

**Comment by naotokui** [comment]
Same here. What version of Ableton Live do you use?  It seems LiveOSC doesn't work with Live 9.

**Comment by samim23** [comment]
Ableton Live 9.5 Suite

**Comment by achimkoh** [comment]
I had the same problem and found that switching 
`remotehost = '127.0.0.1', remoteport=9000, localhost='127.0.0.1', localport=9001` to `remotehost = '127.0.0.1', remoteport=9001, localhost='127.0.0.1', localport=9000` 
in LiveOSC2's LO2OSC.py file gets me past the `server error 9904` error and gives connection to Ableton, i.e. I can call `live.Set()` and run functions like `play()` and `stop()`. 

However, functions that involve an OSC query such as `scan()` seem to fail with `*** timed out waiting for server response`. Not sure what to do from here, any help would be appreciated! 

I'm running python 2.7.13 and Live 9 on MacOS Sierra.

**Comment by samim23** [comment]
Sorry but not sure what kind of black magic I did on my local machine to get pyLive running. 
Probably easiest to strip out pyLive and replace it with a python OSC lib which works on your machine, and then just sends standard OSC to live.

**Comment by achimkoh** [comment]
I found something! So apparently [LiveOSC's secure sockets setting needs to be commented out](http://livecontrol.q3f.org/ableton-liveapi/liveosc/#comment-2511402199) in order to work with Live 9.6+. While I still can't figure out why LiveOSC2 won't return messages to me, this fix gives me LiveOSC in Ableton Live's settings; then your script works fine as is. Although line 70 (```set.save(config["songname"])```) initially gave an error, so I needed to make a subdirectory named "save". 😃 Time to play around!
